pkg/shlex/shlex_test.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2017-2020 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package shlex
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+)
+
+func TestArgv(t *testing.T) {
+	for i, tt := range []struct {
+		desc string
+		in   string
+		want []string
+	}{
+		{
+			desc: "nothing to do",
+			in:   "stuff",
+			want: []string{"stuff"},
+		},
+		{
+			desc: "split",
+			in:   "stuff more stuff",
+			want: []string{"stuff", "more", "stuff"},
+		},
+		{
+			desc: "escape",
+			in:   "stuff\\ more stuff",
+			want: []string{"stuff more", "stuff"},
+		},
+		{
+			desc: "quote",
+			in:   "stuff var='more stuff'",
```

Comment:
Add test for forgetting to close the single quotes.

Reply:
done!